### ISSUE TITLE
[cinder-csi-plugin] Multiple Openstack Cluster Handling 

### ISSUE THREAD

**Issue by sebastienmusso** [kubernetes/cloud-provider-openstack#2056]
/kind question

We have a kubernetes cluster with nodes spread in several Openstack cluster completely autonomous.
We would like to deploy cinder-csi-plugin in order to have cinder pvc handling in 3 distinct openstack cluster with different endpoints url,
or if not possible deploy 3 cinder-csi-plugin in these 3 different az in the same openstack cluster.

Second solution does not seem possible because of hardcoded driverName in the csi plugin.

what do you recommend in my situation ? use an abstraction layer (ceph, portworx) ?

**Comment by jichenjc** [comment]
I think you might want to join discussion 
https://github.com/kubernetes/cloud-provider-openstack/issues/2035

though the title are different, I think both looking for a solution that cinder in multiple region (multiple endpoint)

**Comment by mdbooth** [comment]
Is this possible to implement in OpenStack? If the worker is running on a Nova server in OpenStack A I don't think we can connect a cinder volume from OpenStack B to it. Is this different for multi-region?

@stephenfin who would be a good person to ask about this?

**Comment by stephenfin** [comment]
To the best of my knowledge, this isn't possible in OpenStack. Nova and Cinder need to do a little dance to connect a volume, which requires Nova be able to talk to Cinder. By default, it does using credentials in the `[cinder]` group of options, as documented [here](https://docs.openstack.org/nova/latest/configuration/config.html#cinder). As you can see from that, Nova is using the service catalog by default to look services and their API endpoints up and that's a per deployment thing. You could override this but then you'd lose the ability to talk to any other deployment. I also don't think it would work.

In short, if you want this feature, both Nova and Cinder will need RFEs. We can't do this ourselves.

**Comment by mdbooth** [comment]
Reading the original description again, I wonder if I misunderstood. Are you saying that you have a single k8s spanning multiple openstack clusters and, e.g.:

* Workers in OpenStack A configure cinder CSI to use OpenStack A
* Workers in OpenStack B configure cinder CSI to use OpenStack B

**Comment by dulek** [comment]
> Reading the original description again, I wonder if I misunderstood. Are you saying that you have a single k8s spanning multiple openstack clusters and, e.g.:
> 
> * Workers in OpenStack A configure cinder CSI to use OpenStack A
> * Workers in OpenStack B configure cinder CSI to use OpenStack B

That's my understanding too. To me it's an interesting concept, but it might have more gotchas than we think. Also it does sounds bold that someone has such a cluster already.

**Comment by jichenjc** [comment]
#2035 has some related discussion

to my understanding this can be either 2 openstack clusters A B which are totally separated then the CPO want to consume resources from those 2 clusters  or there is multi-region openstack so A B share keystone but nova/cinder are indpendent 
I think CPO lack of support for those 2 cases 

I think we should not think about VM in A consume volume in B that's not a valid use case from my understanding
